gen3/bin/kube-setup-secrets.sh

```diff
@@ -0,0 +1,280 @@
+#
+# Note that kube.tf cat's this file into ${vpc_name}_output/kube-services.sh,
+# but can also run this standalone if the environment is
+# properly configured.
+#
+set -e
+
+_KUBE_SETUP_SECRETS=$(dirname "${BASH_SOURCE:-$0}")  # $0 supports zsh
+source "${_KUBE_SETUP_SECRETS}/../lib/kube-setup-init.sh"
+
+mkdir -p "${WORKSPACE}/${vpc_name}/apis_configs"
+
+if [[ -f "${WORKSPACE}/${vpc_name}_output/creds.json" ]]; then # update secrets
+  #
+  # Setup the files that will become secrets in "${WORKSPACE}/$vpc_name/apis_configs"
+  #
+  cd "${WORKSPACE}"/${vpc_name}_output
+ 
+  # Note: look into 'kubectl replace' if you need to replace a secret
+  if ! g3kubectl get secrets/indexd-secret > /dev/null 2>&1; then
+    g3kubectl create secret generic indexd-secret --from-file=local_settings.py="${GEN3_HOME}/apis_configs/indexd_settings.py" "--from-file=${GEN3_HOME}/apis_configs/config_helper.py"
+  fi
+  if ! g3kubectl get secret indexd-creds > /dev/null 2>&1; then
+    credsFile=$(mktemp -p "$XDG_RUNTIME_DIR" "creds.json_XXXXXX")
+    jq -r .indexd < creds.json > "$credsFile"
+    g3kubectl create secret generic indexd-creds "--from-file=creds.json=${credsFile}"
+  fi
+
+  if [[ ! -f "${WORKSPACE}"/${vpc_name}/apis_configs/user.yaml ]]; then
+    # user database for accessing the commons ...
+    cp "${GEN3_HOME}/apis_configs/user.yaml" "${WORKSPACE}"/${vpc_name}/apis_configs/
+  fi
+
+  cd "${WORKSPACE}"/${vpc_name}
+fi
+
+if ! g3kubectl get configmaps global > /dev/null 2>&1; then
+  if [[ -f "${WORKSPACE}/${vpc_name}/00configmap.yaml" ]]; then
+    g3kubectl apply -f "${WORKSPACE}/${vpc_name}/00configmap.yaml"
+  else
+    echo "ERROR: unable to configure global configmap - missing ${WORKSPACE}/${vpc_name}/00configmap.yaml"
+    exit 1
+  fi
+fi
+if ! g3kubectl get configmap config-helper > /dev/null 2>&1; then
+  g3kubectl create configmap config-helper --from-file "${GEN3_HOME}/apis_configs/config_helper.py"
+fi
+
+if [[ -f "${WORKSPACE}/${vpc_name}_output/creds.json" ]]; then # update fence secrets
+  if [ ! -d "${WORKSPACE}/${vpc_name}" ]; then
+    echo "${WORKSPACE}/${vpc_name} does not exist"
+    exit 1
+  fi
+
+  cd "${WORKSPACE}/${vpc_name}_output"
+  
+  if ! g3kubectl get secret fence-creds > /dev/null 2>&1; then
+    credsFile=$(mktemp -p "$XDG_RUNTIME_DIR" "creds.json_XXXXXX")
+    jq -r .fence < creds.json > "$credsFile"
+    g3kubectl create secret generic fence-creds "--from-file=creds.json=${credsFile}"
+  fi
+
+  cd "${WORKSPACE}/${vpc_name}"
+  # Generate RSA private and public keys.
+  # TODO: generalize to list of key names?
+  mkdir -p jwt-keys
+  mkdir -p ssh-keys
+
+  if [ ! -f jwt-keys/jwt_public_key.pem ]; then
+    openssl genrsa -out jwt-keys/jwt_private_key.pem 2048
+    openssl rsa -in jwt-keys/jwt_private_key.pem -pubout -out jwt-keys/jwt_public_key.pem
+  fi
+
+  if [ ! -f ssh-keys/id_rsa ]; then
+    ssh-keygen -t rsa -b 4096 -C "giangbui0816@gmail.com" -N "" -f ssh-keys/id_rsa
```

Comment:
Any particular reason to create personalized certs instead of copying them from templates?